### PR TITLE
Gradle, Build scripts, Maven Plugin, NDK update. 

### PR DESCRIPTION
**Pull request by ArtsemKurantsou**
List of updates:
- Support for 16KB page sizes for native lib (Android 15 support)
- NDK update to 27
- min SDK set to 21 and targetSdk set to 35
- AGP 8.5.1 and Gradle 8.9 update along with gradle wrapper
- Build scripts migration from Groovy to Kotlin
- Version management moved to Version Catalog and libraries updated
- Unsupported kotlin-android-extensoin plugin replaced with ViewBinding 
- Publishing migration to the standard maven-publish

### CODE REVIEW

**Comment by scottyab** [comment]
This looks great.  I am away ATM but will review this upon return next week.  We really value the effort here. 👌

File: app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+android {
+    namespace = "com.scottyab.rootbeer.sample"
+    compileSdk = libs.versions.android.compile.sdk.get().toInt()
+    buildToolsVersion = libs.versions.android.build.tools.get()
+
+    defaultConfig {
+        applicationId = "com.scottyab.rootbeer.sample"
+        minSdk = libs.versions.android.min.sdk.get().toInt()
+        targetSdk = libs.versions.android.target.sdk.get().toInt()
+        versionName = "0.9"
```
Comment (scottyab) [inline app/build.gradle.kts:15]: We're just missing `versionCode = 11` otherwise the generated .apk is `RootBeerSample-0.9-[null]-release.apk`

**Comment by scottyab** [comment]
It all pretty much checks out ✅ , again huge thanks! this saved us a bunch of time and sets us up to update the library.  

Last remaining task for me is to confirm the Maven publishing to Nexus works with the updated plugin (and I still remember how to do it 😉 ). Then we can merge and I'll look at the other PRs (long overdue) and publish an updated version of the lib. The Sample app will take a bit more effort due to new Playstore requirments but this can be a seperate task.

File: app/build.gradle.kts (hunk 1 above)
Comment (ArtsemKurantsou) [reply app/build.gradle.kts]: Fixed both versionName and versionCode, now they are specified using properties. As far as I understood it was the initial intention, please correct me if I'm wrong.

**Comment by ArtsemKurantsou** [comment]
> It all pretty much checks out ✅ , again huge thanks! this saved us a bunch of time and sets us up to update the library.
> 
> Last remaining task for me is to confirm the Maven publishing to Nexus works with the updated plugin (and I still remember how to do it 😉 ). Then we can merge and I'll look at the other PRs (long overdue) and publish an updated version of the lib. The Sample app will take a bit more effort due to new Playstore requirments but this can be a seperate task.

Thank you very much for review and the lib itself)
Let me know if there is any issues with that, I'm happy to help

**Review by scottyab — APPROVED** [review]
Confirmed Nexus publishing works as expected after I switched to using tokens (not a change that's needed to the code here). Will merge later. Thanks again!

**Comment by scottyab** [comment]
Rebasing the merge rather than the usual squash merge due to the amount of changes here.

**Pull request merged**